App crashes randomly after taking a picture with the camera

# Bug Report

## Problem
After taking a picture, pushing the OK Button (marked below) of camera-plugin crashes the app randomly.
![image](https://user-images.githubusercontent.com/15822428/94282385-0f74c980-ff50-11ea-9bdf-cb21e89d7896.png)


### What is expected to happen?
The picture should be available in the app for further action.
 
### What does actually happen?
The app crashes.

## Information
```
2020-09-25 16:42:08.277 19807-19807/com.domain.app D/AndroidRuntime: Shutting down VM
2020-09-25 16:42:08.281 19807-19807/com.domain.app E/AndroidRuntime: FATAL EXCEPTION: main
    Process: com.domain.app, PID: 19807
    java.lang.RuntimeException: Unable to resume activity {com.domain.app/com.domain.app.MainActivity}: java.lang.RuntimeException: Failure delivering result ResultInfo{who=null, request=34, result=-1, data=null} to activity {com.domain.app/com.domain.app.MainActivity}: java.lang.NullPointerException: filename cannot be null
        at android.app.ActivityThread.performResumeActivity(ActivityThread.java:4626)
        at android.app.ActivityThread.handleResumeActivity(ActivityThread.java:4659)
        at android.app.servertransaction.ResumeActivityItem.execute(ResumeActivityItem.java:52)
        at android.app.servertransaction.TransactionExecutor.executeLifecycleState(TransactionExecutor.java:176)
        at android.app.servertransaction.TransactionExecutor.execute(TransactionExecutor.java:97)
        at android.app.ActivityThread$H.handleMessage(ActivityThread.java:2261)
        at android.os.Handler.dispatchMessage(Handler.java:107)
        at android.os.Looper.loop(Looper.java:237)
        at android.app.ActivityThread.main(ActivityThread.java:8107)
        at java.lang.reflect.Method.invoke(Native Method)
        at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:496)
        at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:1100)
     Caused by: java.lang.RuntimeException: Failure delivering result ResultInfo{who=null, request=34, result=-1, data=null} to activity {com.domain.app/com.domain.app.MainActivity}: java.lang.NullPointerException: filename cannot be null
        at android.app.ActivityThread.deliverResults(ActivityThread.java:5324)
        at android.app.ActivityThread.performResumeActivity(ActivityThread.java:4613)
        at android.app.ActivityThread.handleResumeActivity(ActivityThread.java:4659) 
        at android.app.servertransaction.ResumeActivityItem.execute(ResumeActivityItem.java:52) 
        at android.app.servertransaction.TransactionExecutor.executeLifecycleState(TransactionExecutor.java:176) 
        at android.app.servertransaction.TransactionExecutor.execute(TransactionExecutor.java:97) 
        at android.app.ActivityThread$H.handleMessage(ActivityThread.java:2261) 
        at android.os.Handler.dispatchMessage(Handler.java:107) 
        at android.os.Looper.loop(Looper.java:237) 
        at android.app.ActivityThread.main(ActivityThread.java:8107) 
        at java.lang.reflect.Method.invoke(Native Method) 
        at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:496) 
        at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:1100) 
     Caused by: java.lang.NullPointerException: filename cannot be null
        at android.media.ExifInterface.<init>(ExifInterface.java:1390)
        at org.apache.cordova.camera.ExifHelper.createInFile(ExifHelper.java:56)
        at org.apache.cordova.camera.CameraLauncher.processResultFromCamera(CameraLauncher.java:481)
        at org.apache.cordova.camera.CameraLauncher.onActivityResult(CameraLauncher.java:808)
        at org.apache.cordova.CordovaInterfaceImpl.onActivityResult(CordovaInterfaceImpl.java:159)
        at org.apache.cordova.CordovaActivity.onActivityResult(CordovaActivity.java:361)
        at android.app.Activity.dispatchActivityResult(Activity.java:8294)
        at android.app.ActivityThread.deliverResults(ActivityThread.java:5317)
        at android.app.ActivityThread.performResumeActivity(ActivityThread.java:4613) 
        at android.app.ActivityThread.handleResumeActivity(ActivityThread.java:4659) 
        at android.app.servertransaction.ResumeActivityItem.execute(ResumeActivityItem.java:52) 
        at android.app.servertransaction.TransactionExecutor.executeLifecycleState(TransactionExecutor.java:176) 
        at android.app.servertransaction.TransactionExecutor.execute(TransactionExecutor.java:97) 
        at android.app.ActivityThread$H.handleMessage(ActivityThread.java:2261) 
        at android.os.Handler.dispatchMessage(Handler.java:107) 
        at android.os.Looper.loop(Looper.java:237) 
        at android.app.ActivityThread.main(ActivityThread.java:8107) 
        at java.lang.reflect.Method.invoke(Native Method) 
        at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(Runti
```


### Command or Code
<!-- What command or code is needed to reproduce the problem? -->



### Environment, Platform, Device
Chrashes have first appeared on Samsung Galaxy Tab A (SM-T515) after Android 10 update.
Same device running android 9 seams to work fine.

### Version information

Cordova Packages:

    cli: 10.0.0
        common: 4.0.2
        create: 3.0.0
        lib: 10.0.0
            common: 4.0.2
            fetch: 3.0.0
            serve: 4.0.0

Project Installed Platforms:

    android: 9.0.0

Project Installed Plugins:

    com-darryncampbell-cordova-plugin-intent: 1.3.0
    com.mirasense.scanditsdk.plugin: 5.14.4
    cordova-plugin-app-version: 0.1.9
    cordova-plugin-camera: 5.0.0 (bug appears on older versions too)
    cordova-plugin-device: 2.0.3
    cordova-plugin-file: 6.0.2
    cordova-plugin-network-information: 2.0.2
    cordova-plugin-whitelist: 1.3.4
    cordova-plugin-x-toast: 2.7.2
    phonegap-plugin-barcodescanner: 8.1.0

Environment:

    OS: Microsoft Windows 10 Pro 10.0.18362 (18362) (win32 10.0.18362) x64
    Node: v13.12.0
    npm: 6.14.4


## Checklist

- [x] I searched for existing GitHub issues
- [x] I updated all Cordova tooling to most recent version
- [x] I included all the necessary information above

Maybe we should just focus on Android on this issue.
I think the issue is related to the activity being destroyed when the camera opens. (So it is probably also related to https://github.com/apache/cordova-plugin-camera/issues/696 ).

My first guess is: 
In the onSaveInstanceState the imageFilePath is saved to the Bundle (with the IMAGE_URI key).
But in the onRestoreStateForActivityResult the IMAGE_URI value is set on imageUri. (So there is a switch).
When the activity has been destroyed the imageFilePath is normally set to null.

```
onSaveInstanceState 

if (this.imageUri != null) {
    state.putString(IMAGE_URI_KEY, this.imageFilePath);
}
```

and
```
onRestoreStateForActivityResult 

if (state.containsKey(IMAGE_URI_KEY)) {
    //I have no idea what type of URI is being passed in
    this.imageUri = Uri.parse(state.getString(IMAGE_URI_KEY));
}
```

When processing the result, the plugin uses the null imageFilePath.

![image](https://user-images.githubusercontent.com/33040889/103383039-8bebeb00-4af1-11eb-8891-45407e8590ce.png)

So maybe just storing and restoring imageUri and imageFilePath seperatly in the onSaveInstance and onRestore methods is enough.

This is the stacktrace from my testing app https://github.com/PieterVanPoyer/cordova-camera-plugin-testing-app when the activity in the emulator is configured to be always destroyed when the camera opens.

```
2020-12-30 22:42:41.774 11147-11147/com.pvpoyer.cameraplugintesting E/AndroidRuntime: FATAL EXCEPTION: main
    Process: com.pvpoyer.cameraplugintesting, PID: 11147
    java.lang.RuntimeException: Unable to resume activity {com.pvpoyer.cameraplugintesting/com.pvpoyer.cameraplugintesting.MainActivity}: java.lang.RuntimeException: Failure delivering result ResultInfo{who=null, request=34, result=-1, data=Intent {  }} to activity {com.pvpoyer.cameraplugintesting/com.pvpoyer.cameraplugintesting.MainActivity}: java.lang.NullPointerException: filename cannot be null
        at android.app.ActivityThread.performResumeActivity(ActivityThread.java:4444)
        at android.app.ActivityThread.handleResumeActivity(ActivityThread.java:4476)
        at android.app.servertransaction.ResumeActivityItem.execute(ResumeActivityItem.java:52)
        at android.app.servertransaction.TransactionExecutor.executeLifecycleState(TransactionExecutor.java:176)
        at android.app.servertransaction.TransactionExecutor.execute(TransactionExecutor.java:97)
        at android.app.ActivityThread$H.handleMessage(ActivityThread.java:2066)
        at android.os.Handler.dispatchMessage(Handler.java:106)
        at android.os.Looper.loop(Looper.java:223)
        at android.app.ActivityThread.main(ActivityThread.java:7656)
        at java.lang.reflect.Method.invoke(Native Method)
        at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:592)
        at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:947)
     Caused by: java.lang.RuntimeException: Failure delivering result ResultInfo{who=null, request=34, result=-1, data=Intent {  }} to activity {com.pvpoyer.cameraplugintesting/com.pvpoyer.cameraplugintesting.MainActivity}: java.lang.NullPointerException: filename cannot be null
        at android.app.ActivityThread.deliverResults(ActivityThread.java:5015)
        at android.app.ActivityThread.performResumeActivity(ActivityThread.java:4431)
        at android.app.ActivityThread.handleResumeActivity(ActivityThread.java:4476) 
        at android.app.servertransaction.ResumeActivityItem.execute(ResumeActivityItem.java:52) 
        at android.app.servertransaction.TransactionExecutor.executeLifecycleState(TransactionExecutor.java:176) 
        at android.app.servertransaction.TransactionExecutor.execute(TransactionExecutor.java:97) 
        at android.app.ActivityThread$H.handleMessage(ActivityThread.java:2066) 
        at android.os.Handler.dispatchMessage(Handler.java:106) 
        at android.os.Looper.loop(Looper.java:223) 
        at android.app.ActivityThread.main(ActivityThread.java:7656) 
        at java.lang.reflect.Method.invoke(Native Method) 
        at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:592) 
        at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:947) 
     Caused by: java.lang.NullPointerException: filename cannot be null
        at android.media.ExifInterface.<init>(ExifInterface.java:1498)
        at org.apache.cordova.camera.ExifHelper.createInFile(ExifHelper.java:56)
        at org.apache.cordova.camera.CameraLauncher.processResultFromCamera(CameraLauncher.java:481)
        at org.apache.cordova.camera.CameraLauncher.onActivityResult(CameraLauncher.java:828)
        at org.apache.cordova.CordovaInterfaceImpl.onActivityResult(CordovaInterfaceImpl.java:159)
        at org.apache.cordova.CordovaActivity.onActivityResult(CordovaActivity.java:361)
        at android.app.Activity.dispatchActivityResult(Activity.java:8310)
        at android.app.ActivityThread.deliverResults(ActivityThread.java:5008)
        at android.app.ActivityThread.performResumeActivity(ActivityThread.java:4431) 
        at android.app.ActivityThread.handleResumeActivity(ActivityThread.java:4476) 
        at android.app.servertransaction.ResumeActivityItem.execute(ResumeActivityItem.java:52) 
        at android.app.servertransaction.TransactionExecutor.executeLifecycleState(TransactionExecutor.java:176) 
        at android.app.servertransaction.TransactionExecutor.execute(TransactionExecutor.java:97) 
        at android.app.ActivityThread$H.handleMessage(ActivityThread.java:2066) 
        at android.os.Handler.dispatchMessage(Handler.java:106) 
        at android.os.Looper.loop(Looper.java:223) 
        at android.app.ActivityThread.main(ActivityThread.java:7656) 
        at java.lang.reflect.Method.invoke(Native Method) 
        at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:592) 
        at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:947) 
```

Hi @PieterVanPoyer 

test step: 
1
```
cordova plugin remove cordova-plugin-camera
cordova plugin add https://github.com/apache/cordova-plugin-camera.git
cordova plugin ls
```
2
```
cordova-plugin-camera 5.0.2-dev "Camera"
cordova-plugin-whitelist 1.3.4 "Whitelist"
```
**### The problem appeared after four or five pictures had been taken**

**Information**

```
2021-01-11 14:33:52.224 12169-12169/com.pvpoyer.cameraplugintesting E/AndroidRuntime: FATAL EXCEPTION: main
    Process: com.pvpoyer.cameraplugintesting, PID: 12169
    java.lang.RuntimeException: Unable to resume activity {com.pvpoyer.cameraplugintesting/com.pvpoyer.cameraplugintesting.MainActivity}: java.lang.RuntimeException: Failure delivering result ResultInfo{who=null, request=34, result=-1, data=null} to activity {com.pvpoyer.cameraplugintesting/com.pvpoyer.cameraplugintesting.MainActivity}: java.lang.NullPointerException: filename cannot be null
        at android.app.ActivityThread.performResumeActivity(ActivityThread.java:4626)
        at android.app.ActivityThread.handleResumeActivity(ActivityThread.java:4659)
        at android.app.servertransaction.ResumeActivityItem.execute(ResumeActivityItem.java:52)
        at android.app.servertransaction.TransactionExecutor.executeLifecycleState(TransactionExecutor.java:176)
        at android.app.servertransaction.TransactionExecutor.execute(TransactionExecutor.java:97)
        at android.app.ActivityThread$H.handleMessage(ActivityThread.java:2261)
        at android.os.Handler.dispatchMessage(Handler.java:107)
        at android.os.Looper.loop(Looper.java:237)
        at android.app.ActivityThread.main(ActivityThread.java:8107)
        at java.lang.reflect.Method.invoke(Native Method)
        at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:496)
        at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:1100)
     Caused by: java.lang.RuntimeException: Failure delivering result ResultInfo{who=null, request=34, result=-1, data=null} to activity {com.pvpoyer.cameraplugintesting/com.pvpoyer.cameraplugintesting.MainActivity}: java.lang.NullPointerException: filename cannot be null
        at android.app.ActivityThread.deliverResults(ActivityThread.java:5324)
        at android.app.ActivityThread.performResumeActivity(ActivityThread.java:4613)
        at android.app.ActivityThread.handleResumeActivity(ActivityThread.java:4659) 
        at android.app.servertransaction.ResumeActivityItem.execute(ResumeActivityItem.java:52) 
        at android.app.servertransaction.TransactionExecutor.executeLifecycleState(TransactionExecutor.java:176) 
        at android.app.servertransaction.TransactionExecutor.execute(TransactionExecutor.java:97) 
        at android.app.ActivityThread$H.handleMessage(ActivityThread.java:2261) 
        at android.os.Handler.dispatchMessage(Handler.java:107) 
        at android.os.Looper.loop(Looper.java:237) 
        at android.app.ActivityThread.main(ActivityThread.java:8107) 
        at java.lang.reflect.Method.invoke(Native Method) 
        at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:496) 
        at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:1100) 
     Caused by: java.lang.NullPointerException: filename cannot be null
        at android.media.ExifInterface.<init>(ExifInterface.java:1390)
        at org.apache.cordova.camera.ExifHelper.createInFile(ExifHelper.java:56)
        at org.apache.cordova.camera.CameraLauncher.processResultFromCamera(CameraLauncher.java:481)
        at org.apache.cordova.camera.CameraLauncher.onActivityResult(CameraLauncher.java:828)
        at org.apache.cordova.CordovaInterfaceImpl.onActivityResult(CordovaInterfaceImpl.java:159)
        at org.apache.cordova.CordovaActivity.onActivityResult(CordovaActivity.java:361)
        at android.app.Activity.dispatchActivityResult(Activity.java:8294)
        at android.app.ActivityThread.deliverResults(ActivityThread.java:5317)
        at android.app.ActivityThread.performResumeActivity(ActivityThread.java:4613) 
        at android.app.ActivityThread.handleResumeActivity(ActivityThread.java:4659) 
        at android.app.servertransaction.ResumeActivityItem.execute(ResumeActivityItem.java:52) 
        at android.app.servertransaction.TransactionExecutor.executeLifecycleState(TransactionExecutor.java:176) 
        at android.app.servertransaction.TransactionExecutor.execute(TransactionExecutor.java:97) 
        at android.app.ActivityThread$H.handleMessage(ActivityThread.java:2261) 
        at android.os.Handler.dispatchMessage(Handler.java:107) 
        at android.os.Looper.loop(Looper.java:237) 
        at android.app.ActivityThread.main(ActivityThread.java:8107) 
        at java.lang.reflect.Method.invoke(Native Method) 
        at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:496) 
        at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:1100)
```

--- a/src/android/CameraLauncher.java
+++ b/src/android/CameraLauncher.java
@@ -92,6 +92,7 @@ public class CameraLauncher extends CordovaPlugin implements MediaScannerConnect
     private static final String GET_All = "Get All";
     private static final String CROPPED_URI_KEY = "croppedUri";
     private static final String IMAGE_URI_KEY = "imageUri";
+    private static final String IMAGE_FILE_PATH_KEY = "imageFilePath";
 
     private static final String TAKE_PICTURE_ACTION = "takePicture";
 
@@ -1350,6 +1351,10 @@ public class CameraLauncher extends CordovaPlugin implements MediaScannerConnect
             state.putString(IMAGE_URI_KEY, this.imageFilePath);
         }
 
+        if(this.imageFilePath != null) {
+            state.putString(IMAGE_FILE_PATH_KEY, this.imageFilePath);
+        }
+
         return state;
     }
 
@@ -1373,6 +1378,10 @@ public class CameraLauncher extends CordovaPlugin implements MediaScannerConnect
         if (state.containsKey(IMAGE_URI_KEY)) {
             //I have no idea what type of URI is being passed in
             this.imageUri = Uri.parse(state.getString(IMAGE_URI_KEY));
+        }
+
+        if (state.containsKey(IMAGE_FILE_PATH_KEY)) {
+           this.imageFilePath = state.getString(IMAGE_FILE_PATH_KEY);
         }
 
         this.callbackContext = callbackContext;

--- a/src/android/CameraLauncher.java
+++ b/src/android/CameraLauncher.java
@@ -1351,7 +1351,7 @@ public class CameraLauncher extends CordovaPlugin implements MediaScannerConnect
             state.putString(IMAGE_URI_KEY, this.imageFilePath);
         }
 
-        if(this.imageFilePath != null) {
+        if (this.imageFilePath != null) {
             state.putString(IMAGE_FILE_PATH_KEY, this.imageFilePath);
         }
 

